### PR TITLE
Fixes to Makefiles, added standard DESTDIR + fixed man pages

### PR DESCRIPTION
**Pull request by leenaars**
The Makefile now respects DESTDIR and PREFIX directives.
The Makefile for building man pages now works.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
 	@echo '# NOTE: You may need to "make testdata" for some tool/* programs'
 	@echo '#'
 
 install: all
-	@$(foreach dir,$(BUILDDIRS),make -C '$(dir)' install && ) echo Installed
+	@$(foreach dir,$(BUILDDIRS),make -e DESTDIR=$(DESTDIR) PREFIX=$(PREFIX) -C '$(dir)' install && ) echo Installed
 
 uninstall:
-	@$(foreach dir,$(BUILDDIRS),make -C '$(dir)' uninstall && ) echo Uninstalled
+	@$(foreach dir,$(BUILDDIRS),make -e -C '$(dir)' uninstall && ) echo Uninstalled
```
Comment (vanrein) [inline Makefile]: Since this is a toplevel taget, shouldn't this also define DESTDIR and PREFIX?

File: Makefile [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
 
 clean:
-	@$(foreach dir,$(BUILDDIRS),make -C '$(dir)' clean && ) echo Cleaned
+	@$(foreach dir,$(BUILDDIRS),make -e -C '$(dir)' clean && ) echo Cleaned
 	@echo '#'
 	@echo '# NOTE: Kept key material, use "make distclean" if you REALLY want to clean it'
 	@echo '#'
```
Comment (vanrein) [inline Makefile]: Since this is a toplevel taget, shouldn't this also define DESTDIR and PREFIX?

File: Makefile [hunk 3]
```diff
@@ -1,22 +1,23 @@
...
 all:
-	@$(foreach dir,$(BUILDDIRS),make -C '$(dir)' all && ) echo Built all
+	@$(foreach dir,$(BUILDDIRS),make DESTDIR=$(DESTDIR) PREFIX=$(PREFIX) -C '$(dir)' all && ) echo Built all
 	@echo '#'
 	@echo '# NOTE: You may need to "make testdata" for some tool/* programs'
 	@echo '#'
 
 install: all
-	@$(foreach dir,$(BUILDDIRS),make -C '$(dir)' install && ) echo Installed
+	@$(foreach dir,$(BUILDDIRS),make -e DESTDIR=$(DESTDIR) PREFIX=$(PREFIX) -C '$(dir)' install && ) echo Installed
```
Comment (vanrein) [inline Makefile]: Why do you need to override *all* envvars with `-e`?  That could lead to problems such as build instabilities; it is usually better to rely on `+=` and `if` etcetera.

File: Makefile (hunk 1 above)
Comment (vanrein) [inline Makefile]: Why do you need to override *all* envvars with `-e`?  That could lead to problems such as build instabilities; it is usually better to rely on `+=` and `if` etcetera.

File: Makefile [hunk 4]
```diff
@@ -1,22 +1,23 @@
...
 
 uninstall:
-	@$(foreach dir,$(BUILDDIRS),make -C '$(dir)' uninstall && ) echo Uninstalled
+	@$(foreach dir,$(BUILDDIRS),make -e -C '$(dir)' uninstall && ) echo Uninstalled
 
 clean:
-	@$(foreach dir,$(BUILDDIRS),make -C '$(dir)' clean && ) echo Cleaned
+	@$(foreach dir,$(BUILDDIRS),make -e -C '$(dir)' clean && ) echo Cleaned
```
Comment (vanrein) [inline Makefile]: Why do you need to override *all* envvars with `-e`?  That could lead to problems such as build instabilities; it is usually better to rely on `+=` and `if` etcetera.

File: Makefile [hunk 5]
```diff
@@ -1,22 +1,23 @@
-
+DESTDIR ?= 
+PREFIX ?= /usr/local/ 
 BUILDDIRS=src lib tool doc test
 
 .PHONEY: all install clean distclean
 
 all:
-	@$(foreach dir,$(BUILDDIRS),make -C '$(dir)' all && ) echo Built all
+	@$(foreach dir,$(BUILDDIRS),make DESTDIR=$(DESTDIR) PREFIX=$(PREFIX) -C '$(dir)' all && ) echo Built all
```
Comment (vanrein) [inline Makefile]: I never saw the form "make VAR=$(VAR)" -- can you point me to the GNU Makefile documentation describing this? (I looked, but failed to find it.)

File: Makefile (hunk 3 above)
Comment (vanrein) [inline Makefile]: I never saw the form "make VAR=$(VAR)" -- can you point me to the GNU Makefile documentation describing this? (I looked, but failed to find it.)

**Comment by vanrein** [comment]
Thanks Michiel.  It would be great to have Nix packages!  I do have a few comments on the commit's coding style, see the commit-inlined comments for that.

In addition, I think the form `$(DESTDIR)$(PREFIX)` is a bit too awkward for anything that is not Nix.  Instead, how would you feel about?

    make -C $(dir) DESTDIR='' PREFIX="$(DESTDIR)$(PREFIX)"

That way, the usual form `$(PREFIX)` occurs in the production rules of the Makefile, which makes them easier to read/digest for people unaware of the specifics of Nix.  This form can safely be used in top- and sublevel Makefiles because `$(DESTDIR)` is reset to an empty string, which can be neutrally prefixed to `$(PREFIX)` in subdirectories.

**Comment by vanrein** [comment]
Michiel, there's one other thing that I don't understand...

Nix installs packages in `/var/nix/23094820934820934820394`, which is basically its package-specifix prefix, right?  Isn't it possible to invoke `make PREFIX=/var/nix/23094820934820934820394 install` on an unmodified TLS Pool build system?  In terms of the environment variables that you mentioned, this may look like `make PREFIX=$(DESTDIR)$(PREFIX) install` -- and would require no changes to the TLS Pool code.

This is not to reject Nix but to push it out of here; it is generally good from a maintenance perspective to avoid merging knowledge from one tool into another, and if I'm correct this is possible.

I'm open to be corrected if I've overlooked anything though!

**Comment by leenaars** [comment]
I think the current commit should be ready to be merged.

File: src/Makefile [hunk 6]
```diff
@@ -38,8 +38,8 @@ clean:
 anew: clean all
 
 install: all
-	install $(TARGETS) "$(PREFIX)/sbin/"
+	install $(TARGETS) "$(DESTDIR)$(PREFIX)/bin/"
```
Comment (vanrein) [inline src/Makefile]: sorry -- changing this is not your prerogative

File: src/Makefile [hunk 7]
```diff
@@ -38,8 +38,8 @@ clean:
...
 
 uninstall:
-	@($foreach t,$(TARGETS),rm -f '$(PREFIX)/sbin/$t' && ) echo Removed TLS Pool daemon
+	@($foreach t,$(TARGETS),rm -f '$(DESTDIR)$(PREFIX)/bin/$t' && ) echo Removed TLS Pool daemon
```
Comment (vanrein) [inline src/Makefile]: sorry -- changing this is not your prerogative

**Comment by vanrein** [comment]
this is almost ready; I do not agree that you changed sbin to bin; we discussed this and I explained why it is in sbin

**Comment by leenaars** [comment]
Ah, yes, indeed. Forgot to fix that. Is fixed now.

**Comment by vanrein** [comment]
I also found the [$(DESTDIR)](http://www.gnu.org/prep/standards/html_node/DESTDIR.html#DESTDIR) habit which was new to me.
Thanks, Michiel!

**Pull request merged**